aimmo-game/simulation/effects.py

```diff
@@ -0,0 +1,49 @@
+from abc import ABCMeta, abstractmethod
+
+
+class _Effect(object):
+    __metaclass__ = ABCMeta
+
+    def __init__(self, avatar):
+        self._avatar = avatar
+
+    @abstractmethod
+    def turn(self):
+        raise NotImplementedError()
+
+
+class _TimedEffect(_Effect):
+    __metaclass__ = ABCMeta
+    EFFECT_TIME = 10
+
+    def __init__(self, *args):
+        super(_TimedEffect, self).__init__(*args)
+        self._time_remaining = self.EFFECT_TIME
+
+    def remove(self):
+        self._avatar.effects.remove(self)
+
+    def turn(self):
+        self._time_remaining -= 1
+        return self._time_remaining != 0
+
+
+class InvulnerabilityEffect(_TimedEffect):
+    def __init__(self, *args):
+        super(InvulnerabilityEffect, self).__init__(*args)
+        self._avatar.resistance += 1000
```

Comment:
What happens if resistance is altered later on by something else? For example:
- start => resistance 5
- acquire InvulnerabilityEffect => resistance 1005
- Get hit by a 'lose resistance' object (this doesn't exist yet) => resistance 995
- expire InvulnerabilityEffect => resistance -5